reduction_scripts/reduce_data.py

```diff
@@ -916,43 +916,99 @@ def run_save_3dcube(metadata, prev_suffix, curr_suffix, **args):
     # INICIATE THE SCRIPT
     # --------------------------------------------
 
-    # Parse command line arguments
-    if len(sys.argv) < 2:
-        print("Error: Please enter a raw data directory.")
-        sys.exit(1)
+    # Initialize ArgumentParser with a description
+    parser = argparse.ArgumentParser(
+        description="The Python data reduction pipeline for WiFeS."
+    )
+
+    # The raw data directory is a required positional argument
+    parser.add_argument("data_dir", type=str, help="Path to the raw data directory.")
+
+    # Option for specifying the path to the red parameters JSON file
+    parser.add_argument(
+        "--red-params",
+        type=str,
+        help="Optional: Path to the configuration JSON file containing parameters for reducing the blue arm.",
+    )
+
+    # Option for specifying the path to the blue parameters JSON file
+    parser.add_argument(
+        "--blue-params",
+        type=str,
+        help="Optional: Path to the configuration JSON file containing parameters for reducing the blue arm.",
+    )
+
+    args = parser.parse_args()
 
-    # Read the raw data directory from command line
-    data_dir = os.path.abspath(sys.argv[1]) + "/"
+    # Validate and process the data_dir
+    data_dir = os.path.abspath(args.data_dir)
+    if not data_dir.endswith("/"):
+        data_dir += "/"
+    print(f"Processing data in directory: {data_dir}")
 
-    naxis2_to_process = 0
-    if len(sys.argv) == 3:
-        naxis2_to_process = int(sys.argv[2])
+    # Handling redcution parameters.
+
+    params_path = {
+        "blue": None,
+        "red": None,
+    }
+
+    # Red
+    if args.red_params:
+        params_path["red"] = os.path.abspath(args.red_params)
+        print(f"Using red parameters from: {params_path['red']}")
+
+    # Blue
+    if args.blue_params:
+        params_path["blue"] = os.path.abspath(args.blue_params)
+        print(f"Using blue parameters from: {params_path['blue']}")
 
     # Classify all raw data (red and blue arm)
+    naxis2_to_process = 0  # TODO is this used in half frame (stellar mode)? Check that and include it as a parameter in the json files if needed.
     obs_metadatas = classify(data_dir, naxis2_to_process)
 
     # Set paths
     reduction_scripts_dir = os.path.dirname(__file__)
     working_dir = os.getcwd()
 
+    # Check grism and observing mode in the first science image of each arm
+    sci_filename_red = obs_metadatas["red"]["sci"][0]["sci"][0] + ".fits"
+    sci_filename_blue = obs_metadatas["blue"]["sci"][0]["sci"][0] + ".fits"
+
+    # Grism: grating are expected to be diferent on each arm. Read them into a dictionary.
+    grism = {
+        "blue": pyfits.getheader(data_dir + sci_filename_blue)["GRATINGB"],
+        "red": pyfits.getheader(data_dir + sci_filename_red)["GRATINGR"],
+    }
+
+    # Observing mode: should be the same in both arm
```

Comment:
There's no need to check that the observing mode is the same in both arms as the instrument set-up does not allow for different observing modes in each arm. 
In the code below, you have missed the conditional for `is_subnodshuffle()` so this piece of code does not work as is. I think it'd be better to leave it as it was (inside the loop) so in case there's no blue or red arm data, the pipeline keeps running smoothly and the conditionals are more simple. 
In case you prefer to leave this before the loop, please double-check that it works for data 1 arm only.

Reply:
Same here. I introduced this check to prevent local users from inadvertently mixing data from different modes. Should we keep it?